### DISCUSSION TITLE
Any interest in developing an app for Freediving?

### DISCUSSION THREAD

**Discussion by Sinama** [zepp-health/zeppos-forum#279] [User Feedback]
I have an Amazfit GTS 2 coming soon in the mail.  My wife already got hers.  There are 90+ workouts available on the watch and none of them are freediving/apnea.  Here are some helpful things that a smart watch could do better than a cellphone.

-O2 Tables & CO2 Tables - User first does a max breath hold attempt.  Then the app builds O2 tables and CO2 tables based on the user's max breath.  When the user feels their first contraction they tap the watch face or the button.  After that every 30 secs are announced by the watch.  The watch counts down from 10 when to breath before breath hold and also counts down from 10 before taking breath again during resting period.
-Time alarms - User wears the watch while diving.  They choose from the app the depth they are attempting and what freediving event they are doing.  The watch then buzzes automatically at times recommended to take for reaching certain depths along the way.  This way the freediver can gauge if they are making good timing or if they are too fast or too slow to descend.  Could possibly be done with an alarm sound if the hand of freediver used for equalizing is the one with the watch on it.
-Surface interval calculator - Diver tells watch the depth they are attempting.  The watch senses when it is submerged (can it do this) and then counts until it is out of the water again. Following that it sets an alarm to alert the freediver when it is safe to dive again.
-During breath holds of the static tables above, the watch monitors heartrate and announces it at certain intervals of the breath hold.
-I have seen that the watch only measures up to 80% SpO2.  If this can be pushed downwards any farther it would be helpful to freedivers.  If the watch could measure and announce when a freediver is reaching certain concentrations of SpO2 that would help the diver understand how quickly they are depleting their oxygen during a dive.

I don't have the developer talent to create this, but maybe there is a developer here who might take an interest.
I imagine that some freedivers might also Google looking for the same mini-app for Zepp that I am.  If you do, maybe you could bump this message so that people know there is a demand for it.

**Comment by htoooth** [comment]
‌‌‌‌GTS2 probably cannot develop a mini-program. It is not an equipment installed with Zepp OS.  @Sinama
